spring-cloud-stream-app-documentation-maven-plugin/src/main/java/org/springframework/cloud/stream/app/documentation/plugin/ConfigurationMetadataDocumentationMojo.java

```diff
@@ -89,7 +90,8 @@ public void execute() throws MojoExecutionException {
 			}
 
 			ScatteredArchive archive = new ScatteredArchive(mavenProject);
-			ClassLoader classLoader = metadataResolver.createClassLoader(archive);
+			BootClassLoaderFactory bootClassLoaderFactory = new BootClassLoaderFactory(archive, null);
```

Comment:
No we can't really, without compromising the scope of it: we need one here to eg list enum values, while want the one created by the the properties listing to be managed by that method entirely. The whole point of a recent change was to make sure that the creation (and disposal) of the CL was internal to the listing, to make sure that we don't leak them.
This plugin is really an exception, so it's not a big deal IMHO